### ISSUE TITLE
Door catch issue

### ISSUE THREAD

**Issue by revk** [revk/SolarSystem#149]
Some types of "catch" lock may need to be treated differently to a mag lock or motorised lock.

The issue is that an unlock will only be "cleared" by the door opening from closed, so if unlock done when door already open, it means closing will not lock.

This could be an option, or perhaps a general change as it may be OK with other lock types. Solonoid and motorised locks usually have their own detection of door close to avoid operating when the door is open. Do we need to cater for when this is not the case.

Basically...

Now: Unlock applied when unlock needed and maintained until door opened and then closed again.

Required: Unlock applied only if currently closed, and applied until door is opened. All other cases unlock is not applied.

**Comment by revk** [comment]
Actually this is not as simple, we used the output state for unlock to control the lock state (ie locking) I think. But may be ok as long as door open also considered. Need to carefully check the lock state stuff.

Also need to consider case of no open input so unlocked for a time period only.

**Comment by revk** [comment]
I have decided to make this a config option.
